numpy/core/src/common/simd/sse/math.h

```diff
@@ -174,4 +174,33 @@ NPY_FINLINE npyv_s64 npyv_min_s64(npyv_s64 a, npyv_s64 b)
     }
 #endif
 
+// trunc
+#ifdef NPY_HAVE_SSE41
+    #define npyv_trunc_f32(A) _mm_round_ps(A, _MM_FROUND_TO_ZERO)
+    #define npyv_trunc_f64(A) _mm_round_pd(A, _MM_FROUND_TO_ZERO)
+#else
+    NPY_FINLINE npyv_f32 npyv_trunc_f32(npyv_f32 a)
+    {
+        const npyv_f32 szero = _mm_set1_ps(-0.0f);
+        npyv_s32 roundi = _mm_cvttps_epi32(a);
+        npyv_f32 trunc = _mm_cvtepi32_ps(roundi);
+        // respect signed zero, e.g. -0.5 -> -0.0
+        npyv_f32 rzero = _mm_or_ps(trunc, _mm_and_ps(a, szero));
+        // if overflow return a
+        return npyv_select_f32(_mm_cmpeq_epi32(roundi, _mm_castps_si128(szero)), a, rzero);
+    }
+    NPY_FINLINE npyv_f64 npyv_trunc_f64(npyv_f64 a)
+    {
+        const __m128d szero = _mm_set1_pd(-0.0);
+        const __m128d one = _mm_set1_pd(1.0);
+        const __m128d two_power_52 = _mm_set1_pd(0x10000000000000);
+        const __m128d nonsign_bit = _mm_castsi128_pd(_mm_set1_epi64x(0x7fffffffffffffff));
```

Comment:
```suggestion
        const __m128d nonsign_bit = _mm_castsi128_pd(npyv_setall_s64(0x7fffffffffffffffLL));
 ```
buggy on msvc(32-bit), see:
https://github.com/numpy/numpy/blob/a7e22741d058a4c1c8d6b93facbafdf712806614/numpy/core/src/common/simd/simd.h#L30-L48 

you can just use `npyv_abs_f64` instead too.